models/users.js

```diff
@@ -450,6 +450,36 @@ const getUsersBasedOnFilter = async (query) => {
   return [];
 };
 
+const updateRoles = async (userId, newRoles) => {
+  try {
+    const userDoc = await userModel.doc(userId).get();
+    const user = userDoc.data();
+    let roles;
+    if (Object.keys(newRoles).includes("member")) {
+      if (newRoles.member === true) {
+        if (user?.roles?.member) {
+          return { isRoleUpdated: false };
+        }
+      } else if (newRoles.member === false) {
+        if (user?.roles?.member === false) {
+          return { isRoleUpdated: false };
+        }
+      }
+      roles = user.roles ? { ...user.roles, member: newRoles.member } : { member: newRoles.member };
```

Comment:
can we initialize `roles` in the begging itself with ` ...user.roles` so that here we just need to add an extra property

Reply:
You mean I have to write like this:- 
`roles = {...userData.roles} ? { ...userData.roles, archived: newRoles.archived } : { archived: newRoles.archived };`